DCHdigi/src/DCHdigi.cpp

```diff
@@ -0,0 +1,671 @@
+        double smearing_z = m_gauss_z_cm( m_engine );
+        if( m_create_debug_histos.value() ) hSz->Fill( smearing_z );
+
+        hit_projection_on_the_wire +=  smearing_z*(wire_direction_ez.Unit());
+        if( m_create_debug_histos.value() )
+        {
+            // the distance from the hit projection and the wire should be zero
+            TVector3 dummy_vector = this->Calculate_hitpos_to_wire_vector(ilayer, nphi,hit_projection_on_the_wire);
+            hDww->Fill( dummy_vector.Mag() );
+        }
+
+        //       smear position perpendicular to the wire
+        double smearing_xy = m_gauss_xy_cm( m_engine );
+        if( m_create_debug_histos.value() ) hSxy->Fill( smearing_xy );
+        float distanceToWire_real = hit_to_wire_vector.Mag();
+
+        // protect against negative values
+        float distanceToWire_smeared = std::max(0.0, distanceToWire_real +smearing_xy );
+
+        std::int32_t type = 0;
+        std::int32_t quality = 0;
+        float eDepError =0;
+        // length units back to mm
+        auto positionSW   = Convert_TVector3_to_EDM4hepVector(hit_projection_on_the_wire, 1./MM_TO_CM );
+        auto directionSW  = Convert_TVector3_to_EDM4hepVector(wire_direction_ez         , 1./MM_TO_CM );
+        float distanceToWire = distanceToWire_smeared/MM_TO_CM;
+
+        auto [clusterCount,clusterSize] = CalculateClusters(input_sim_hit);
+
+        extension::MutableDriftChamberDigiV2 oDCHdigihit;
+        oDCHdigihit.setCellID(input_sim_hit.getCellID());
+        oDCHdigihit.setType(type);
+        oDCHdigihit.setQuality(quality);
+        oDCHdigihit.setTime(input_sim_hit.getTime());
+        oDCHdigihit.setEDep(input_sim_hit.getEDep());
+        oDCHdigihit.setEDepError(eDepError);
+        oDCHdigihit.setPosition(positionSW);
+        oDCHdigihit.setDirectionSW(directionSW);
+        oDCHdigihit.setDistanceToWire(distanceToWire);
+        oDCHdigihit.setClusterCount(clusterCount);
+        oDCHdigihit.setClusterSize(clusterSize);
+
+        output_digi_hits.push_back(oDCHdigihit);
+
+        extension::MutableMCRecoDriftChamberDigiV2Association oDCHsimdigi_association;
+        oDCHsimdigi_association.setDigi( oDCHdigihit  );
+        oDCHsimdigi_association.setSim( input_sim_hit  );
+
+        }// end loop over hit collection
+
+    /////////////////////////////////////////////////////////////////
+    return std::make_tuple<colltype_out,colltype_out2>(std::move(output_digi_hits),std::move(output_digi_sim_association));
+}
+
+///////////////////////////////////////////////////////////////////////////////////////
+///////////////////////       finalize       //////////////////////////////////////////
+///////////////////////////////////////////////////////////////////////////////////////
+StatusCode DCHdigi::finalize()
+{
+    if( m_create_debug_histos.value() )
+    {
+        std::unique_ptr<TFile> ofile{TFile::Open ( "dch_digi_alg_debug.root", "recreate" ) };
```

Comment:
Can we have the name of the output rootfile for debug th1's configurable from the steering file? (with a default value so that it can also be ignored)

Reply:
done